pkg/sql/sem/tree/cast.go

```diff
@@ -0,0 +1,414 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package tree
+
+import (
+	"fmt"
+
+	"github.com/lib/pq/oid"
+)
+
+// CastContext specifies how a given Castable can be used.
+// A higher value corresponds to a higher strictness.
+// Higher value CastContext can use lower value CastContexts.
+type CastContext uint8
+
+const (
+	_ CastContext = iota
+	// CastContextExplicit implies that the cast can only be used
+	// in an explicit context.
+	CastContextExplicit
+	// CastContextAssignment implies that the cast can done implicitly
+	// in assign contexts (e.g. on UPDATE and INSERT).
+	CastContextAssignment
+	// CastContextImplicit implies the cast can be set implicitly
+	// in any context.
+	CastContextImplicit
+)
+
+// CastMethod specifies how the cast is done.
+// This is legacy from postgres, but affects type overloads,
+// so we also include it.
+type CastMethod byte
```

Comment:
I don't understand the need for this. Does this affect a cast's behavior at all?